src/Decoder.c

```diff
@@ -4320,6 +4201,208 @@ static ZyanStatus ZydisNodeHandlerMvexE(const ZydisDecodedInstruction* instructi
+            /* BOUND   */ (1 << 4) - 1  //              ignore `.X`
+        };
+        id_rm &= (is_reg ? mask_rm[def_rm] : 0xFF);
+
+        // Commented out for future reference. Not required at the moment as it's always either
+        // a "take all" or "take nothing" situation.
+
+        //static const ZyanU8 mask_ndsndd[ZYDIS_REGKIND_MAX_VALUE + 1] =
+        //{
+        //    /* INVALID */ 0,
+        //    /* GPR     */ (1 << 5) - 1,
+        //    /* X87     */ 0,            // never encoded in `.vvvv`
+        //    /* MMX     */ 0,            // never encoded in `.vvvv`
+        //    /* VR      */ (1 << 5) - 1,
+        //    /* TMM     */ (1 << 5) - 1,
+        //    /* SEGMENT */ 0,            // never encoded in `.vvvv`
+        //    /* TEST    */ 0,            // never encoded in `.vvvv`
+        //    /* CONTROL */ 0,            // never encoded in `.vvvv`
+        //    /* DEBUG   */ 0,            // never encoded in `.vvvv`
+        //    /* MASK    */ (1 << 5) - 1,
+        //    /* BOUND   */ 0             // never encoded in `.vvvv`
+        //};
+    }
+
+    // Validate
+
+    // `.vvvv` is not allowed, if the instruction does not encode a NDS/NDD operand
+    if (!def_ndsndd && context->cache.vvvv)
+    {
+        return ZYDIS_STATUS_BAD_REGISTER;
+    }
+    // `.v'` is not allowed, if the instruction does not encode a NDS/NDD or VSIB operand
+    if (!def_ndsndd && !has_vsib && context->cache.V2)
+    {
+        return ZYDIS_STATUS_BAD_REGISTER;
+    }
+
+    static const ZyanU8 available_regs[2][ZYDIS_REGKIND_MAX_VALUE + 1] =
+    {
+        // 16/32 bit mode
+        {
+            /* INVALID */ 255,
+            /* GPR     */   8,
+            /* X87     */   8,
+            /* MMX     */   8,
+            /* VR      */   8,
+            /* TMM     */   8,
+            /* SEGMENT */   6,
+            /* TEST    */   8,
+            /* CONTROL */   8,
+            /* DEBUG   */   8,
+            /* MASK    */   8,
+            /* BOUND   */   4
+        },
+        // 64 bit mode
+        {
+            /* INVALID */ 255,
+            /* GPR     */  16,
+            /* X87     */   8,
+            /* MMX     */   8,
+            /* VR      */  32,
+            /* TMM     */   8,
+            /* SEGMENT */   6,
+            /* TEST    */   8,
+            /* CONTROL */  16,
+            // Attempts to reference DR8..DR15 result in undefined opcode (#UD) exceptions. DR4 and
+            // DR5 are only valid, if the debug extension (DE) flag in CR4 is set. As we can't
+            // check this at runtime we just allow them.
+            /* DEBUG   */   8,
+            /* MASK    */   8,
+            /* BOUND   */   4
+        }
+    };
+
+    const ZyanBool is_64_bit = (instruction->machine_mode == ZYDIS_MACHINE_MODE_LONG_64);
+    if ((id_reg >= available_regs[is_64_bit][def_reg]) ||
+        (id_ndsndd >= available_regs[is_64_bit][def_ndsndd]) ||
```

Comment:
This check is more restrictive than previous version:
```C
if ((context->decoder->machine_mode == ZYDIS_MACHINE_MODE_LONG_64) &&
    (context->cache.v_vvvv > 7))
{
	return ZYDIS_STATUS_BAD_REGISTER;
}
```
It seems that removal of this behavior is accidental (see my comment above).